server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -469,6 +487,15 @@ public void applyClusterState(ClusterChangedEvent event) {
         }
     }
 
+    private boolean canUpdateInPlace(DiscoveryNodes nodes, RepositoryMetadata updatedMetadata, Repository repository) {
+        if (nodes.getMinNodeVersion().before(UPDATE_IN_PLACE_VERSION)) {
```

Comment:
This looks fragile because the minimum node version in the cluster can go downwards (if `UPDATE_IN_PLACE_VERSION` doesn't ultimately end in `.0.0`). I think it's ok if we're checking this as part of a cluster state update, because a node won't be concurrently rejoining the cluster. Can we assert we're on the master thread at this point to make sure we don't check this wrongly?

Reply:
Actually why do we need the version check? If we update the repository metadata and there's older nodes around then will they break or will they just keep using the old repo metadata for any ongoing shard-level operations?

Reply:
Alrighty, as just discussed. The BwC will be handled via some other mechanism here. Removed the BwC logic from this PR entirely for now as it's not a problem in interaction with versions that aren't already a problem anyway.